### PR TITLE
chore(docs): Clean up dead links

### PR DESCRIPTION
**Pull request by sime**
Based on comments https://www.notion.so/satoshilabs/Broken-Links-docs-FIRMWARE-7fd9e24c7c7a422db16eef6a859de0b4

### CODE REVIEW

**Comment by marnova** [comment]
What I've found in the Notion and it's still not working:
- https://github.com/trezor/trezor-firmware/blob/master/python/README.md?plain=1#L100 should point to https://github.com/trezor/trezor-firmware/tree/master/python/docs
- https://github.com/trezor/trezor-firmware/blob/master/python/README.md?plain=1#L110 -> https://github.com/trezor/trezor-firmware/tree/master/python/tools
- https://github.com/trezor/trezor-firmware/blob/master/docs/hardware/model-one/firmware-format.md?plain=1#L35 and https://github.com/trezor/trezor-firmware/blob/master/docs/hardware/model-one/firmware-format.md?plain=1#L78 -> https://github.com/trezor/trezor-firmware/blob/master/python/src/trezorlib/firmware.py
- https://github.com/trezor/trezor-firmware/blob/master/docs/tests/device-tests.md?plain=1#L106 -> ???
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L7 -> https://github.com/trezor/trezor-firmware/blob/master/ci/environment.yml
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L23 -> https://github.com/trezor/trezor-firmware/blob/master/ci/prebuild.yml
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L54 -> https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L141 -> https://github.com/trezor/trezor-firmware/blob/master/ci/test.yml
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L159 -> https://github.com/trezor/trezor-firmware/blob/master/docs/tests/ui-tests.md
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L187 -> https://github.com/trezor/trezor-firmware/blob/master/docs/tests/click-tests.md
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L193 -> https://github.com/trezor/trezor-firmware/blob/master/docs/tests/upgrade-tests.md
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L231 -> https://github.com/trezor/trezor-firmware/blob/master/ci/test-hw.yml
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L256 -> https://github.com/trezor/trezor-firmware/blob/master/docs/tests/device-tests.md
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L264 -> https://github.com/trezor/trezor-firmware/blob/master/ci/posttest.yml
- https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L273 -> https://github.com/trezor/trezor-firmware/blob/master/ci/deploy.yml

I used links to the files in github, please use relative links where applicable (e.g. for https://github.com/trezor/trezor-firmware/blob/master/docs/ci/jobs.md?plain=1#L193  use `../tests/upgrade-tests.md` instead of what I've posted here - https://github.com/trezor/trezor-firmware/blob/master/docs/tests/upgrade-tests.md).

**Review by sime — COMMENTED** [review]
If I could approve, I would.

File: python/.changelog.d/2191.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed broken links in docs.
```
Comment (mmilata) [inline python/.changelog.d/2191.fixed:1]: Wouldn't add this to changelog - you can silence the failing CI job by adding `[no changelog]` to commit message, or you can ignore the failing job.

**Review by mmilata — COMMENTED** [review]
LGTM, let's drop the changelog entry and merge it.

**Review by mmilata — APPROVED** [review]

**Pull request merged**